extensions/amp-animation/0.1/css-expr-ast.js

```diff
@@ -596,6 +601,111 @@ export class CssTranslateNode extends CssFuncNode {
 
 
 /**
+ * An AMP-specific `width()` and `height()` functions.
+ */
+export class CssDimSizeNode extends CssNode {
+  /**
+   * @param {string} dim
+   * @param {?string=} opt_selector
+   * @param {?string=} opt_selectionMethod Either `undefined` or "closest".
+   */
+  constructor(dim, opt_selector, opt_selectionMethod) {
+    super();
+    /** @const @private */
+    this.dim_ = dim;
+    /** @const @private */
+    this.selector_ = opt_selector || null;
+    /** @const @private */
+    this.selectionMethod_ = opt_selectionMethod || null;
+  }
+
+  /** @override */
+  css() {
+    // No CSS represention.
+    return '';
+  }
+
+  /** @override */
+  isConst() {
+    return false;
+  }
+
+  /** @override */
+  calc(context) {
+    const size =
+        this.selector_ ?
+        context.getElementSize(this.selector_, this.selectionMethod_) :
+        context.getCurrentElementSize();
+    return new CssLengthNode(getDimSide(this.dim_, size), 'px');
+  }
+}
+
+
+/**
+ * An AMP-specific `rand()` function.
+ */
+export class CssRandNode extends CssNode {
+  /**
+   * @param {?CssNode=} left
+   * @param {?CssNode=} right
```

Comment:
It's not fully clear what `left` and `right` mean. It looks like it's the threshold for the `rand()` result but it's not clear how it goes from node -> threshold. Documenting these would be helpful.